src/array/ops/cast.rs

```diff
@@ -701,12 +705,120 @@ impl EmbeddingArray {
 
 impl ImageArray {
     pub fn cast(&self, dtype: &DataType) -> DaftResult<Series> {
-        self.physical.cast(dtype)
+        match dtype {
+            #[cfg(feature = "python")]
+            DataType::Python => {
+                Python::with_gil(|py| {
+                    let mut ndarrays = Vec::with_capacity(self.len());
+                    let mut bitmap = arrow2::bitmap::MutableBitmap::with_capacity(self.len());
```

Comment:
You should be able to use the validity of the data array and clone it

Reply:
Oh duh, good point!